internal/plugin/connectors/tcp/ssl/ssl.go

```diff
@@ -51,18 +51,29 @@ func NewDbSSLMode(o options, requireCanVerifyCAOnly bool) (DbSSLMode, error) {
 
 		// MySQL on the other hand notes in its docs that it ignores
 		// SSL certs if supplied in REQUIRED sslmode.
-		if requireCanVerifyCAOnly && len(o["sslrootcert"]) > 0 {
+		if requireCanVerifyCA && len(o["sslrootcert"]) > 0 {
 			sslMode.VerifyCaOnly = true
 		}
+
 	case "verify-ca":
-		// Skip TLS's own verification: it requires full verification since Go 1.3.
+		// Skip stdlib's verification: it requires full verification since Go 1.3.
 		sslMode.InsecureSkipVerify = true
 		sslMode.VerifyCaOnly = true
-	//case "verify-full":
-	//	sslMode.ServerName = o["host"]
+
+	case "verify-full":
+		// Use stdlib's verification
+		sslMode.InsecureSkipVerify = false
+		sslMode.VerifyCaOnly = false
+
+		// 'sslhost', when not empty, takes precedence over 'host'
+		if o["sslhost"] != "" {
```

Comment:
I'm not sure if this should be an empty string check or a length check. In either case, my follow-up PR will use a structured representation instead, so this shouldn't be an issue.